python/ray/train/_internal/dataset_spec.py

```diff
@@ -141,8 +143,15 @@ def preprocess_datasets(
                     continue
                 if conf.fit:
                     ds_to_fit = datasets[k]
-            if ds_to_fit:
+            if ds_to_fit and prep.fit_status() in (
+                Preprocessor.FitStatus.NOT_FITTED,
+                Preprocessor.FitStatus.PARTIALLY_FITTED,
+            ):
```

Comment:
Is this a controversial change? A loaded preprocessor that's been fitted already shouldn't fit again.